### PR TITLE
🎉 Source Amazon Seller Partner: enable stream attribute customisation from Source configuration

### LINKED ISSUE #18958: Source Amazon Seller Partner: GET_SALES_AND_TRAFFIC_REPORT with incremental sync mode does not update last data
## Environment
- **Airbyte version**:  example is 0.40.18-alpha
- **OS Version / Instance**: AWS EC2
- **Deployment**: Docker env
- **Source Connector and version**: Amazon Seller Partner 0.2.28
- **Destination Connector and version**: n.a.
- **Step where error happened**: Sync job

## Current Behavior
After setting up the sync mode to `Incremental | append` the job correctly runs, but according to Amazon documentation reports may take up to a week to populate data after a reporting periods closes, so with cursor field set to queryEndDate fresh data are missing and the last reports are all with traffic equals 0.

## Expected Behavior
When downloading stream with incremental sync mode the connector should use a rolling time window containing also the last week already processed. 

## Logs
/

## Steps to Reproduce
1. Configure Amazon Selling Partner connector
2. Enable GET_SALES_AND_TRAFFIC_REPORT replication stream with sync mode set to Incremental | Append
3. Verify data in time against Amazon seller central reports

## Are you willing to submit a PR?
I can work on it, but some help would be great

### PR DESCRIPTION
**Pull request by dandpz**
fixes #18958 
replaces #19480 

## What
When downloading report about SALES and TRAFFIC from Amazon Seller Partner it can occur a delay on the data freshness, this is a issue when using incremental sync mode: because of this delay we lose some data and we cannot retrieve again. Hardcoding the sla days is not possible because a delay sla for some reports is not available inside the official documentation of Amazon Seller Partner (i.e. GET_SALES_AND_TRAFFIC_REPORT). 

## How
The solution is to introduce the possibility to add generic overwrite of stream attributes. For the specific issue mentioned in #18958 with this change the user can modify the attribute `availability_sla_days` to the desired value. 

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [x] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.py [hunk 1]
```diff
@@ -108,6 +108,15 @@ class Config:
...
     )
 
+    # This field has been introduced because some users experienced a delay on report data update,
+    # for certain reports the data availability SLA is not explicited this field allows to customize this value. Using this sort of
+    # options make possible to extend the usage adding more customization per stream.
+    advanced_stream_options: str = Field(
+        None,
+        description="Additional information to configure report options. This varies by report type, not every report implement this kind of feature. Must be a valid json string.",
+        examples=['{"GET_SALES_AND_TRAFFIC_REPORT": {"availability_sla_days": 3}}', '{"GET_SOME_REPORT": {"custom": "true"}}'],
+    )
```
Comment (sh4sh) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.py:118]:
```suggestion
        examples=['{"GET_SALES_AND_TRAFFIC_REPORT": {"availability_sla_days": 3}}', '{"GET_SOME_REPORT": {"custom": "true"}}'],
        order=13,
    )
```

**Review by marcosmarxm — APPROVED** [review]
Thanks @dandpz

**Pull request merged**